metakb/transform/pmkb.py

```diff
@@ -0,0 +1,480 @@
+            type="DiseaseDescriptor",
+            label=disease,
+            value=schemas.Disease(id=response['value_object_descriptor']['value']['id'])  # noqa: E501
+        ).dict(exclude_none=True)
+
+        if tissue_types:
+            vod['extensions'] = [
+                {
+                    "type": "Extension",
+                    "name": "tissue_types",
+                    "value": tissue_types
+                }
+            ]
+
+        self.transformed['disease_descriptors'][disease] = vod
+        return [vod]
+
+    def _get_gene_descriptors(self, variant):
+        """Fetch gene descriptors.
+
+        :param Dict variant: harvested PMKB variant object
+        :return: List (len == 1) containing VOD of normalized match, or
+            empty list if normalization fails
+        """
+        symbol = variant['gene']['name']
+        invalid_keys = self.invalid_keys['genes']
+        if symbol in invalid_keys:
+            return []
+
+        gene_descriptors = self.transformed['gene_descriptors']
+        vod = gene_descriptors.get(symbol)
+        if vod:
+            return [vod]
+
+        response, _ = self.vicc_normalizers.normalize_gene([symbol])
+        if not response:
+            logger.warning(f"Gene normalization of {symbol} failed.")
+            invalid_keys.add(symbol)
+            return []
+
+        normalized_id = response['records'][0].concept_id
+
+        vod = schemas.ValueObjectDescriptor(
+            id=f"pmkb.normalize.gene:{symbol}",
+            type="GeneDescriptor",
+            label=symbol,
+            value=schemas.Gene(id=normalized_id)
+        ).dict(exclude_none=True)
+
+        gene_descriptors[symbol] = vod
+        return [vod]
+
+    def _get_variant_descriptors(self, variant, gene_id):
+        """Fetch variant descriptors.
+
+        :param Dict variant: PMKB variant object
+        :param str gene_id: identifier for gene_context field
+        :return: List (len == 1) containing VOD of normalized match, or
+            empty list if normalization fails
+        """
+        label = variant['name']
+        invalid_keys = self.invalid_keys['variants']
+        if label in invalid_keys:
+            return []
+
+        vod = self.transformed['variation_descriptors'].get(label)
+        if vod:
+            return [vod]
+
+        response = self.vicc_normalizers.normalize_variant([label])
+        if not response:
+            logger.warning(f"Variant normalization of {label} failed.")
+            invalid_keys.add(label)
+            return []
+
+        vod = schemas.VariationDescriptor(
+            id=f"pmkb.variant:{variant['id']}",
+            type="VariationDescriptor",
+            label=label,
+            value_id=response.value_id,
+            value=response.value,
+            gene_context=gene_id,
+            molecule_context=response.molecule_context,
+            structural_type=response.structural_type,
+            ref_allele_seq=response.ref_allele_seq,
```

Comment:
Might be wrong because I still get confused on this... I don't think we should be including normalizer responses in fields other than value object fields. If I am wrong, then let me know and we should make these additions to CIViC and MOA.

Reply:
No, I think you have it correct -- the Value comes from the normalizer, but the descriptor refers to the way that the Value is described by the resource.